test/junit/scala/collection/immutable/RangeConsistencyTest.scala

```diff
@@ -172,4 +172,43 @@ class RangeConsistencyTest {
     assert(r.sum(possiblyNotDefaultNumeric) == Int.MinValue)
     assert(nr.sum(possiblyNotDefaultNumeric) == Int.MinValue)
   }
+
+  @Test
+  def test_SI10086()  {
+    implicit val customIntegral =
+      new Numeric.IntIsIntegral with Ordering.IntOrdering {}
+
+    val nr = NumericRange(1, 10, 1)
+    assert(nr.min == 1)
+    assert(nr.max == 9)
+
+    // Also test with custom ordering.
+    assert(nr.min(customIntegral.reverse) == 9)
+    assert(nr.max(customIntegral.reverse) == 1)
+
+    case class A(v: Int)
+
+    implicit object aIsIntegral extends scala.math.Integral[A] {
+      def compare(x: A, y: A): Int = x.v - y.v
```

Comment:
This is inconsistent.  Orderings must be transitive with respect to the sign of `compare` but this is not true with `A(Int.MinValue)`, `A(0)`, and `A(Int.MaxValue)`.  Though extreme values are not used in this test, it is nonobvious that this method fails, and there is no advantage to using subtraction rather than the `compare` method on the underlying Ints.